noir-projects/aztec-nr/aztec/src/context/call_interfaces.nr

```diff
@@ -0,0 +1,407 @@
+use dep::protocol_types::{abis::function_selector::FunctionSelector, address::AztecAddress, traits::Deserialize};
+
+use crate::context::private_context::PrivateContext;
+use crate::context::public_context::PublicContext;
+use crate::context::avm_context::AvmContext;
+use crate::context::gas::GasOpts;
+use crate::context::public_context::FunctionReturns;
+
+use crate::hash::hash_args;
+use crate::oracle::arguments;
+
+struct PrivateCallInterface<T> {
+    target_contract: AztecAddress,
+    selector: FunctionSelector,
+    args_hash: Field,
+}
+
+impl<T> PrivateCallInterface<T> {
+    pub fn call<N>(self, context: &mut PrivateContext) -> T where T: Deserialize<N> {
+        let returns = context.call_private_function_with_packed_args(
+            self.target_contract,
+            self.selector,
+            self.args_hash,
+            false,
+            false
+        );
+        let unpacked: T = returns.unpack_into();
+        unpacked
+    }
+
+    pub fn view<N>(self, context: &mut PrivateContext) -> T where T: Deserialize<N> {
```

Comment:
Here we do the opposite, we call it `view`. I think this is the right way to go about it

Reply:
Because this is user facing 😄